scala/scala_import.bzl

```diff
@@ -20,27 +21,45 @@ def _scala_import_impl(ctx):
         ),
         jars_to_labels = jars2labels,
         providers = [
-            _create_provider(current_jars, transitive_runtime_jars, jars, exports)
+            java_common.merge([jars_provider, deps_provider, runtime_deps_provider, exports_provider])
         ],
     )
-def _create_provider(current_target_compile_jars, transitive_runtime_jars, jars, exports):
-    return java_common.create_provider(
-        use_ijar = False,
-        compile_time_jars = depset(transitive = [current_target_compile_jars, exports.compile_jars]),
-        transitive_compile_time_jars = depset(transitive = [jars.transitive_compile_jars, current_target_compile_jars, exports.transitive_compile_jars]) ,
-        transitive_runtime_jars = depset(transitive = [transitive_runtime_jars, jars.transitive_runtime_jars, current_target_compile_jars, exports.transitive_runtime_jars]) ,
-      )
+
+def _jars_to_provider(jars):
+  providers = []
+  for jar in jars:
+    if JavaInfo in jar:
+      fail("jars must contain only jar files")
+
+    providers.append(_jar_to_provider(jar.files.to_list()[0]))
+
+  return java_common.merge(providers)
+
+def _jar_to_provider(jar):
+  return JavaInfo(
```

Comment:
Why are you building the `JavaInfo` like this (without the `deps`, `exports`,`runtime_deps`)?
I think that's not the correct semantics

Reply:
At this point there is only a jar file - not a `JavaInfo`